scripts/markdown-magic.config.js

```diff
@@ -40,18 +40,15 @@ exports.transforms = {
    */
   toc: (content, options, config) => {
     const IGNORED_HEADINGS_REGEXP = /Features|Table of Contents/i;
-    return (
-      '\n' +
-      markdownToc(config.outputContent, {
-        slugify: require('uslug'),
-        bullets: options.bullets,
-        firsth1: false,
-        // if filter is supplied, maxdepth is apparently ignored,
-        // so we have to do it ourselves.
-        filter: (str, ele) => ele.lvl < 2 && !IGNORED_HEADINGS_REGEXP.test(str)
-      }).content +
-      '\n'
-    );
+    const toc = markdownToc(config.outputContent, {
+      slugify: require('uslug'),
+      bullets: options.bullets,
+      firsth1: false,
+      // if filter is supplied, maxdepth is apparently ignored,
+      // so we have to do it ourselves.
+      filter: (str, ele) => ele.lvl < 2 && !IGNORED_HEADINGS_REGEXP.test(str)
+    }).content;
+    return '\n' + toc + '\n';
```

Comment:
might be worth adding a comment as to the relevance of the `\n`'s.  they ensure markdownlint doesn't barf on the rendered output.

when we ensure that the auto-generated content is no longer under VCS (and is only part of the docs build), then we don't need the extra newlines.

Reply:
I had just considered it good style -- didn't realize an automated tool agreed with me. :)

Reply:
[Implementation preference](https://github.com/mochajs/mocha/pull/3710#issuecomment-470178999)?